packages/browser-destinations/destinations/fullstory/src/identifyUserV2/index.ts

```diff
@@ -0,0 +1,139 @@
+import type { BrowserActionDefinition } from '@segment/browser-destination-runtime/types'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+import type { FS } from '../types'
+import camelCase from 'lodash/camelCase'
+import { segmentEventSource } from '..'
+
+// Change from unknown to the partner SDK types
+const action: BrowserActionDefinition<Settings, FS, Payload> = {
+  title: 'Identify User',
+  description: 'Sets user identity variables',
+  platform: 'web',
+  defaultSubscription: 'type = "identify"',
+  fields: {
+    userId: {
+      type: 'string',
+      required: false,
+      description: "The user's id",
+      label: 'User ID',
+      default: {
+        '@path': '$.userId'
+      }
+    },
+    anonymousId: {
+      type: 'string',
+      required: false,
+      description: "The user's anonymous id",
+      label: 'Anonymous ID',
+      default: {
+        '@path': '$.anonymousId'
+      }
+    },
+    displayName: {
+      type: 'string',
+      required: false,
+      description: "The user's display name",
+      label: 'Display Name',
+      default: {
+        '@path': '$.traits.name'
+      }
+    },
+    email: {
+      type: 'string',
+      required: false,
+      description: "The user's email",
+      label: 'Email',
+      default: {
+        '@path': '$.traits.email'
+      }
+    },
+    traits: {
+      type: 'object',
+      required: false,
+      description: 'The Segment traits to be forwarded to FullStory',
+      label: 'Traits',
+      default: {
+        '@path': '$.traits'
+      }
+    }
+  },
+  perform: (FS, event) => {
+    let newTraits: Record<string, unknown> = {}
+
+    if (event.payload.traits) {
+      newTraits = Object.entries(event.payload.traits).reduce(
+        (acc, [key, value]) => ({
+          ...acc,
+          [camelCaseField(key)]: value
```

Comment:
I don't think we want to mess with field names for V2. WDYT?

Reply:
Seems like something we could remove.

Reply:
Cool, please do!